# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmFrameIter.cpp
# Commit: 9a162cc369e3
# Full Hash: 9a162cc369e3b6f51f5767924e63d46539c4f631
# Author: Julien Pages <jpages@mozilla.com>
# Date: 2025-02-22 09:14:45
# Regressor Bug: 1861078
# File Overlap Count: 1
# Description:
#   Bug 1861078 - wasm: make metadata offsets relative to CodeBlock->codeBase. r=rhunt
#   
#   Metadata used offsets to the beginning of code segment. Change that to use
#   offsets relative to the beginning of CodeBlock. This allows us to avoid offseting
#   metadata later on and prepare the work for simplification of stackmaps.
# ==============================================================================

diff -r e543a77786c9 -r 9a162cc369e3 js/src/wasm/WasmFrameIter.cpp
--- a/js/src/wasm/WasmFrameIter.cpp	Fri Feb 21 17:09:51 2025 +0000
+++ b/js/src/wasm/WasmFrameIter.cpp	Fri Feb 21 17:17:58 2025 +0000
@@ -1340,7 +1340,7 @@
   const CodeBlock* codeBlock = LookupCodeBlock(pc, &codeRange);
   if (codeBlock) {
     code = codeBlock->code;
-    codeBase = codeBlock->segment->base();
+    codeBase = codeBlock->base();
     MOZ_ASSERT(codeRange);
   } else if (!LookupBuiltinThunk(pc, &codeRange, &codeBase)) {
     return false;